packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page.tsx

```diff
@@ -85,10 +94,23 @@ const Page: NextPageWithLayout = () => {
       data.properties.map((property) => {
         const propertyKey = extractBaseUri(property.$id);
 
-        return [propertyKey, { $ref: property.$id }];
+        const prop: ValueOrArray<PropertyTypeReference> = property.array
+          ? {
+              type: "array",
+              minItems: property.minValue,
+              maxItems: property.maxValue,
+              items: { $ref: property.$id },
+            }
+          : { $ref: property.$id };
+
+        return [propertyKey, prop];
       }),
     );
 
+    const required = data.properties
+      .filter((property) => property.required)
+      .map((property) => `${property.$id.split("/v/")[0]}/`); // @todo: is there a better way to do this?
```

Comment:
You can probably convert this all to one `for .. of` loop to just make it clearer/faster (and probably extract it to a function)

Reply:
Done in 90a8589f3c4f0b7b10765e096ecb211ec06533a3. Please let me know if you'd have anything changed.